web_src/js/features/repo-release.ts

```diff
@@ -46,3 +50,57 @@ function initTagNameEditor() {
     hideTargetInput(e.target as HTMLInputElement);
   });
 }
+
+function initGenerateReleaseNotes() {
+  const button = document.querySelector<HTMLButtonElement>('#generate-release-notes');
+  if (!button) return;
+
+  const tagNameInput = document.querySelector<HTMLInputElement>('#tag-name');
+  const targetInput = document.querySelector<HTMLInputElement>("input[name='tag_target']");
+  const previousTagSelect = document.querySelector<HTMLSelectElement>('#release-previous-tag');
+  const missingTagMessage = button.getAttribute('data-missing-tag-message') || 'Tag name is required';
+  const generateUrl = button.getAttribute('data-generate-url');
+
+  button.addEventListener('click', async () => {
+    const tagName = tagNameInput.value.trim();
+
+    if (!tagName) {
+      showErrorToast(missingTagMessage);
+      tagNameInput?.focus();
+      return;
+    }
+
+    const form = new URLSearchParams();
+    form.set('tag_name', tagName);
+    form.set('tag_target', targetInput.value || '');
+    form.set('previous_tag', previousTagSelect.value || '');
+
+    button.classList.add('loading', 'disabled');
+    try {
+      const resp = await POST(generateUrl, {
+        data: form,
+      });
+      const data = await resp.json();
+      if (!resp.ok) {
+        throw new Error(data.errorMessage || resp.statusText);
+      }
+
+      previousTagSelect.value = data.previous_tag;
```

Comment:
```suggestion

      if (!resp.ok) {
        throw new Error(data.errorMessage || resp.statusText);
      }

      const data = await resp.json();
      previousTagSelect.value = data.previous_tag;
```

No need to parse the json when it's not going to be used in case of throw.

Reply:
But it's not right and won't work.

If you don't parse JSON, then you won't have `data` , then `data.errorMessage` is always undefined.

Reply:
Oops, I missed that it accesses `data` in the error message, sorry.

Reply:
reverted it below.